boa_engine/src/object/jsmap.rs

```diff
@@ -164,7 +164,7 @@ impl JsMap {
     ///
     /// let some_object = JsArray::new(context);
     ///
-    /// let js_map = JsMap::from_object(some_object, context).unwrap();
+    /// let js_map = JsMap::from_object(some_object.into(), context).unwrap();
```

Comment:
Should panic is a bit of a last resort when testing if something must throw an error. I'd recommend using `is_err` with an `assert!` instead :)
Something like: `assert!(JsMap::from_object(some_object.into(), context).is_err());`